src/Database/Franz/Server.hs

```diff
@@ -151,19 +151,24 @@ startServer Settings{..} = withFranzReader livePrefix $ \franzReader -> do
             -- just start a session without thinking about archives
             Nothing -> respondLoop path
             -- Mount a squashfs image and increment the counter
-            Just prefix | src <- prefix </> path -> withSharedResource vMounts path
-              (mountFuse src (livePrefix </> path))
-              (\fuse -> do
-                -- close the last client's streams
-                streams <- atomically $ do
-                  streams <- readTVar $ vStreams franzReader
-                  writeTVar (vStreams franzReader) $ HM.delete path streams
-                  pure streams
-                forM_ (HM.lookup path streams) $ mapM_ closeStream
-                killFuse fuse (livePrefix </> path) `finally` do
-                  pid <- getPid fuse
-                  forM_ pid $ \p -> logServer ["Undead squashfuse detected:", show p])
-              (const $ respondLoop path)
+            Just prefix | src <- prefix </> path -> do
+              -- ^ check if an archive exists
+              exist <- doesFileExist src
```

Comment:
I don't really understand why this is a desirable behaviour: it is very surprising that user passes in `archivePrefix` but it's not necessarily used. It seems akin to `readFile` returning an empty file content instead of throwing an exception if the file is missing: very strange. It seems very conceivable that user makes a typo in `archivePrefix` and end up with the wrong thing happening completely (and fully silently at that).

I think it makes no real sense to do this and definitely not quietly at start time and without documentation with reasoning in code...

Reply:
This is to allow streams to be archived transparently (those who were reading live streams can read the stream with the same path). I'm open to suggestions

Reply:
We should at least add a non-default flag for this like `--allow-missing-archives` with a description of what it will do. Further, we should output a message here saying something like "archivePath doesn't exist, making a fresh archive at src" or the like.

Reply:
I don't understand... It doesn't create any fresh archive. Are you suggesting to disable live streams entirely by default?

Reply:
Intended behavior is not clear from provided documentation / comments / code.

Reply:
I'm suggesting that it's surprising as a user to pass a flag that explicitly says "use this archive path" and the software ignores it without a single bit of indication about it

Reply:
I'm not sure why you are saying `archivePrefix` is not used. It's always used

Reply:
> It's always used

This is the only place it's used and if the path doesn't exist, it's silently ignored and that's surprising. Normally if some software takes a flag that says "use file XYZ" and XYZ doesn't exist, the software quits or if it has some fallback, is very vocal about that.

If user makes a typo in `archivePrefix`, they won't be able to tell that something didn't go as expected. Which is why I'm suggesting a loud message if archive is not found and/or a flag that enables/disables this behaviour.

Reply:
Would it display a loud message on the server side depending on a path client specified?

Reply:
Sure: anything is better than nothing. The help menu on archivePath should document this too

Reply:
I see. Updated the code